### PR TITLE
Add: PostgreSQL scanner

### PR DESCRIPTION
**Pull request by GAZDOWN**

### CODE REVIEW

File: src/actors/containerization/services/postgresql/postgresql-scan/scan.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+
+if __name__ == "__main__":
+    augeas_input = json.load(sys.stdin)
+
+    instances = [] 
+
+    # aug_postgresql might be undefined, that usually means the postgresql data
+    # were not initialized.
+    try:
```
Comment (mfranczy) [inline src/actors/containerization/services/postgresql/postgresql-scan/scan.py]: I think this exception is too wide. 

Origin of the failure it's not obvious, we don't know if we couldn't find ```aug_postgresql``` which means postgresql data weren't initialized, or data were initialized but we couldn't find some specific key which means something... 

Tomorrow we have demo, so I suggest to improve this in the next iteration, although right now i recommend to send message from KeyError exception to stderr (I mean original message from KeyError + short explanation what happened)

File: src/actors/containerization/services/postgresql/postgresql-scan/scan.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            "properties": augeas_hba["properties"]
+        }
+
+        instances.append(instance)
+
+    except KeyError as e:
+        pass
+
+    if len(instances):
+        print(json.dumps({"postgresql_scanner": [{"instances": instances}]}))
```
Comment (bertinatto) [inline src/actors/containerization/services/postgresql/postgresql-scan/scan.py]: Perhaps you should remove the `if` clause and always print this.

If no PostgreSQL installation was found, IMO we should print an empty list.

File: src/actors/containerization/services/postgresql/postgresql-scan/scan.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+        instance["postgresql_conf"] = {
+            "path": augeas_postgresql["absolute_path"],
+            "properties": augeas_postgresql["properties"]
+        }
+
+        instance["pg_hba_conf"] = {
+            "path": augeas_hba["absolute_path"],
+            "properties": augeas_hba["properties"]
+        }
```
Comment (bertinatto) [inline src/actors/containerization/services/postgresql/postgresql-scan/scan.py]: You might want to do something like this instead:

```python
instance = {
    'postgresql_conf' : {
        'path': augeas_postgresql['absolute_path'],
        'properties': augeas_postgresql['properties']
    },
    'pg_hba_conf' : {
        'path': augeas_hba['absolute_path'],
        'properties': augeas_hba['properties']
    },        
print(json.dumps({"postgresql_scanner": [{"instances": [instance]}]}))
```

**Comment by examon** [comment]
leapp-ci build

**Comment by examon** [comment]
leapp-ci build

**Comment by vojtechsokol** [comment]
leapp-ci build

**Comment by vojtechsokol** [comment]
leapp-ci build

**Comment by vinzenz** [comment]
outdated

**Pull request closed without merging**